app/api/controllers/RequestHandler.scala

```diff
@@ -123,21 +126,34 @@ object RequestHandler {
 
       implicit class Response(result: Result) {
 
+        private val HttpDateFormatter = DateTimeFormatter
+          .ofPattern("EEE, dd MMM yyyy HH:mm:ss z", Locale.ENGLISH)
+          .withZone(ZoneId.of("GMT"))
+
         def withApiHeaders(correlationId: String, responseHeaders: (String, String)*)(implicit appConfig: AppConfig, apiVersion: Version): Result = {
           val maybeDeprecatedHeader =
             if (appConfig.isApiDeprecated(apiVersion))
               List(
-                "Deprecation" -> s"This endpoint is deprecated. See the API documentation: ${appConfig.apiDocumentationUrl}"
+                "Deprecation" -> s"${HttpDateFormatter.format(appConfig.deprecatedOn(apiVersion).get)}",
```

Comment:
Using `.get` can be risky - what if the deprecatedOn returns none? could cause a runtime exception which would be problematic

Reply:
Might also be worth creating a private function called formatConfig and pass in the config value

Reply:
NEVAAAHHHH have "option.get" in production code 🙁 

Tests are ok under certain circumstances, but "get" does rather invite unexpected runtime side-effects.

Reply:
basically just replace the if with sm like appConfig.deprecatedOn(apiVersion).map(deprecatedDate => ...).toList